Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
```

Comment:
>  but it appears you're unwilling to do the same

That is because you haven't convince me on such issue yet. 

> If you'd rather, I'll just stopping reviewing this PR and you can let someone else do it.

That's definitely not my intention. Since writing singleton on my own is not too hard. I will go ahead on this approach like below:
```ruby
def self.instance
  @instance ||= CoreFormulaTap.new
end
```

Apology if I offend you in anyway.